src/main/osd/osd_elements.c

```diff
@@ -206,6 +206,36 @@ int osdConvertTemperatureToSelectedUnit(int tempInDegreesCelcius)
 }
 #endif
 
+static void osdFormatVoltage(char * buff, int batteryVoltage)
```

Comment:
It would be much cleaner to implement function to print float with required number of digits (not limited to 2 decimal places) and call it from here.

Reply:
Yes, but i think there is no another use of such complex function. I working at simplified implementation of formating.

Reply:
Another option is to fix printf to support width and precision (I have it in my local repository), add float support (lot of code implements it's own way to handle fractional numbers) and that just calculate required parameters for printf ... 

Problem with current printf is that it's quite ugly implementation ... it will be probably better to discard it completely and use use another one ...

Reply:
As you can see i dropped support of first implementation using tfp_sprintf and current formating is procedural, simplified as much as posible.

Reply:
@ledvinap: Yes, I think that would be preferable - consistently using a good implementation of `printf` is much preferred over having individual bits of code using `itoa()` in inconsistent form. Do you want to open a pull request for this change? (Another option would be to use `printf` provided by https://github.com/mpaland/printf/blob/master/printf.c.)